### PR TITLE
Support LEF58_EOLKEEPOUT rule

### PR DESCRIPTION
**Pull request by osamahammad21**
* Support in OpenDB (
  * codeGenerator
  * lefin
  * TestLef58Properties
* Support in TritonRoute
  * frLayer and frConstraint
  * io
  * gc/ FlexGC_eol
  * gcTest

closes [[OR-985]](https://openroad.atlassian.net/browse/OR-985)

[OR-985]: https://openroad.atlassian.net/browse/OR-985

### CODE REVIEW

**Comment by osamahammad21** [comment]
@maliberty this fails because of pdngen test which has KEEPOUT rule with CORNERONLY difined. Should I support that in db and drt or just update the ok files with db warnings.

**Comment by maliberty** [comment]
@osamahammad21 it is there in ORFS's asap7 setup so I imagine it will show up there too as an issue.  How hard is it to support?

**Comment by osamahammad21** [comment]
Not hard. It is just checking that the rectangle has at least one corner in the query region.

File: src/OpenDB/test/cpp/TestLef58Properties.cpp [hunk 1]
```diff
@@ -124,6 +124,16 @@ BOOST_AUTO_TEST_CASE( test_default )
     BOOST_TEST(layer->isRectOnly()==true);
     BOOST_TEST(layer->isRectOnlyExceptNonCorePins()==true);
 
+    auto keepoutRules = layer->getTechLayerEolKeepOutRules();
+    BOOST_TEST(keepoutRules.size() == 1);
+    auto keepoutRule = (odb::dbTechLayerEolKeepOutRule*) *(keepoutRules.begin());
+    BOOST_TEST(keepoutRule->getEolWidth() == 0.2*distFactor);
```
Comment (maliberty) [inline src/OpenDB/test/cpp/TestLef58Properties.cpp:130]: clang-format

**Review by maliberty — COMMENTED** [review]
> Not hard. It is just checking that the rectangle has at least one corner in the query region.

Ok then please do so.

**Review by maliberty — APPROVED** [review]

**Pull request merged**